lib/wpcom-error-handler/wpcom-error-handler.php

```diff
@@ -82,7 +82,8 @@ function wpcom_get_error_backtrace( $last_error_file, $last_error_type, $for_irc
  * When we run at shutdown we must not die as then the pretty printing of the Error doesn't happen which is lame sauce.
  */
 function wpcom_custom_error_handler( $whether_i_may_die, $type, $message, $file, $line ) {
-	if ( ! ( $type & ini_get( 'error_reporting' ) ) ) {
+	// phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.runtime_configuration_error_reporting
+	if ( ! is_numeric( $type ) || ! ( $type & error_reporting() ) ) { 
```

Comment:
@GaryJones makes a good point. `is_numeric()` returns true even for numeric strings like "1", "3.55444", etc... whereas `is_int()` only returns true for numbers. See [is_int#Examples](https://www.php.net/manual/en/function.is-int.php#refsect1-function.is-int-examples).
```suggestion
	if ( ! is_int( $type ) || ! ( $type & error_reporting() ) ) { 
```

Reply:
@brandon-m-skinner @GaryJones I actually went with `is_numeric()` because bitwise operations still work with numeric strings but I agree, being more specific here is probably a better idea